projects/ad9361/src/ad9361.c

```diff
@@ -572,35 +572,29 @@ const char *ad9361_ensm_states[] = {
  * @param num The number of bytes to read.
  * @return 0 in case of success, negative error code otherwise.
  */
-int32_t ad9361_spi_readm(struct spi_device *spi, uint32_t reg,
+int32_t ad9361_spi_readm(struct spi_desc *spi, uint32_t reg,
 			 uint8_t *rbuf, uint32_t num)
 {
-	uint8_t buf[2];
-	int32_t ret;
+	int32_t ret = 0;
 	uint16_t cmd;
-
+	uint8_t *rbuffer;
 	if (num > MAX_MBYTE_SPI)
 		return -EINVAL;
 
 	cmd = AD_READ | AD_CNT(num) | AD_ADDR(reg);
-	buf[0] = cmd >> 8;
-	buf[1] = cmd & 0xFF;
+	rbuffer = (uint8_t *) malloc(num + 2);
```

Comment:
There should be no need to cast the return of the malloc as the void pointer is safely promoted to any other type of pointer.

Reply:
Also there might be some value in verifying that the malloc call succeeded by:
<code>
if(!rbuffer)
    return -1;
</code>
To avoid the risk of errors from trying to free a NULL pointer of to assign values to it.